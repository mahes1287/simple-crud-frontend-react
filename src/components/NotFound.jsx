import React from "react";
import { Link } from "react-router-dom";
export default function NotFound() {
  return (
    <div className="container mx-auto p-10 text-2xl">
      <div data-testid="not-found">Sorry, The page is not found.....</div>
      <Link to={"/"} className="underline">
        Home
      </Link>
    </div>
  );
}
