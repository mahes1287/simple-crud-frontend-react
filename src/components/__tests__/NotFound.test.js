import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import NotFound from "../NotFound";
import renderer from "react-test-renderer";
describe("Not found page ::", () => {
  test(`should give the "Sorry, The page is not found....." text in the page`, () => {
    render(<NotFound />, { wrapper: BrowserRouter });
    const textElement = screen.getByText("Sorry, The page is not found.....");
    expect(textElement).toBeInTheDocument();
  });

  test("Should give the Link to Home", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <NotFound />
        </BrowserRouter>
      )
      .toJSON();

    render(<NotFound />, { wrapper: BrowserRouter });
    const linkElement = screen.getByRole("link", { name: "Home" });
    expect(linkElement).toBeInTheDocument();
    expect(linkElement.getAttribute("href")).toMatchInlineSnapshot(`"/"`);
  });
});
