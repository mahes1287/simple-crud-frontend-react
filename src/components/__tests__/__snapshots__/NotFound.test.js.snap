// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not found page :: Should give the Link to Home 1`] = `
<div
  className="container mx-auto p-10 text-2xl"
>
  <div
    data-testid="not-found"
  >
    Sorry, The page is not found.....
  </div>
  <a
    className="underline"
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
</div>
`;
